archivers/wayback_archiver.py

```diff
@@ -1,81 +1,91 @@
-import time, requests, os
+import time, requests
+
+from loguru import logger
 from bs4 import BeautifulSoup
 
 from storages import Storage
 from .base_archiver import Archiver, ArchiveResult
-
-from loguru import logger
+from configs import WaybackConfig
 
 
 class WaybackArchiver(Archiver):
     name = "wayback"
 
-    def __init__(self, storage: Storage, driver):
+    def __init__(self, storage: Storage, driver, config: WaybackConfig):
         super(WaybackArchiver, self).__init__(storage, driver)
+        self.config = config
+        # TODO: this logic should live at the auto-archiver level
         self.seen_urls = {}
 
     def download(self, url, check_if_exists=False):
-        if check_if_exists and url in self.seen_urls:
-            return self.seen_urls[url]
+        if check_if_exists:
+            if url in self.seen_urls: return self.seen_urls[url]
+
+            logger.debug(f"checking if {url=} already on archive.org")
```

Comment:
With this archiver in particular, I think it needs to look at the time since it's last been archived. For some test URLs I was trying out, this was pulling back Wayback results from months ago, which I think is not desired behavior, because the page could have changed quite a bit. For other archivers this is much less of an issue.

Reply:
I've been setting check_if_exists to false and accepting the performance hit throughout all the archivers. Not sure if this is the issue here. Albeit I've posted some URL's 10 times to archive.org and hit their daily limit.